### ISSUE TITLE
Feature request: console log signed integers with description

### ISSUE THREAD

**Issue by paulrberg** [nomiclabs/hardhat#1440]
I love the console.log functionality that you guys offer, but it's limited in scope to unsigned integers (I think `uint256`, specifically). I am in need of debugging a contract that works with signed numbers alone, so I resorted to using `console.logInt`:

```solidity
console.logInt(x.log2());
```

But I am logging multiple vars and it's becoming harder to know which is which in the console output. It'd be great if I could do this:

```solidity
console.logInt("x.log2", x.log2());
```

Or even before, ditch the `logInt` altogether and overload the usual `console.log` to add support for signed integers.

**Comment by fvictorio** [comment]
I think there is a reason why plain `console.log` doesn't support signed integers (I guess related to distinguishing a call to `console.log(uint(3))` from one to `console.log(int(3))`), but I don't know the details.

Assuming that's the case and we can't make the second option, the request is to add these methods, right?

```solidity
function logInt(int, int);
function logInt(int, int, int);
function logInt(int, int, int, int);
```

**Comment by alcuadrado** [comment]
Unfortunately, solidity would refuse to compile a version of `console.log()` that takes `int` and `uint`, because `console.log(1)` would be ambiguous.

I think adding an optional label to all the the `console.logX()` is a great idea, @paulrberg !

**Comment by alcuadrado** [comment]
Now that I think of, maybe that wasn't your idea haha

What do you mean by `console.logInt("x.log2", x.log2());`?

**Comment by paulrberg** [comment]
Re-reading my post, it does seem a bit ambiguous. Sorry guys! What I meant is that it would be great to be able to add a string label as an optional first parameter. Like so:

```solidity
int256 foo = 100;
console.logInt("this is foo", foo);
```

I understand why we can't use `console.log` for both unsigned and signed integers.

**Comment by alcuadrado** [comment]
No worries. I somehow got it on the first read and then thought I was wrong lol

This is a great idea, and really easy to implement.

If someone wants to take it, you just need to modify [this script](https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-core/scripts/console-library-generator.js) to generate more overloads in `console.sol` and their ts counterpart, and then run it.

It's actually a good first issue, and one of the few of those that actually implements something new instead of fixing a bug.

**Comment by alcuadrado** [comment]
Hey @Bidon15, I saw your comment in #1441, so I thought about tagging you here. This is a very nice first issue, that can help you better understand how hardhat network's console.log works. It's pretty hacky and cool.

**Comment by LeonardoSalvucci** [comment]
Hi @alcuadrado! I was reviewing this issue for making this improvement and i notice that it's easy to add "int" to permutations but a little tricky to add tests to verify this change. Is there some documentation about testing and how to add/configure json files to add this new tests? or some good practice to not making it in the wrong way?

**Comment by fvictorio** [comment]
Hey @LeonardoSalvucci, I completely missed your comment, sorry :disappointed:

I think this issue should wait until https://github.com/NomicFoundation/hardhat/issues/2178 is ready. Let me know if you want to work in another `good-first-issue`-labeled issue and I'll help.

**Comment by gabrielkrell** [comment]
Yes, this is a real head-scratcher. Hardhat documentation for console.log makes no mention of logInt (in the [tutorial](https://hardhat.org/tutorial/debugging-with-hardhat-network) or the [real documentation](https://hardhat.org/hardhat-network/docs/overview#console.log)) and console.log is one of the things you expect to "just work" for basic types like ints. implementing my own `log(message, int, int)` is something I had to do when I was using truffle.

tldr bump
